lua/autorun/server/sv_restrict_propcore.lua

```diff
@@ -1,4 +1,41 @@
 -- Propcore is allowed to everyone, but functions in the restrictedFunctions array will be restricted to devotee+ only
+local file_name = "cfc_propcore_whitelist"
+local whitelistedPlayers = whitelistedPlayers or {}
+
+if not file.Exists( file_name .. ".txt", "DATA") then
+    --Creating new data file
+    file.Write( file_name .. ".txt", "" )
+else
+    local fileContents = file.Read( file_name .. ".txt" )
+    local translated = util.JSONToTable( fileContents )
+    whitelistedPlayers = translated or {}
+end
+
+local function saveWhitelistChanges()
+    local translated = util.TableToJSON( whitelistedPlayers, true )
+
+    file.Write( file_name .. ".txt", translated )
+end
+
+function addPlayersToPropcoreWhitelist( players )
+    for _, ply in pairs( players ) do
+        table.insert( whitelistedPlayers, ply:SteamID() )
+    end
+
+    saveWhitelistChanges()
+end
+
+function removePlayersFromPropcoreWhitelist( players )
+    for _, ply in pairs( players ) do
+        table.RemoveByValue( whitelistedPlayers, ply:SteamID() )
+    end
+
+    saveWhitelistChanges()
+end
+
+function isPlayerWhitelisted( ply )
```

Comment:
I think this should be renamed` playerIswhitelisted`

Also, if we're going to be making global functions, we should make a table to put them under so we're not polluting the global namespace

Reply:
Good point will do.